docs/highlighters/main.md

```diff
@@ -55,6 +55,7 @@ This highlighter defines the following styles:
 * `dollar-double-quoted-argument` - parameter expansion inside double quotes (`$foo` inside `""`)
 * `back-double-quoted-argument` -  backslash escape sequences inside double-quoted arguments (`\"` in `"foo\"bar"`)
 * `back-dollar-quoted-argument` -  backslash escape sequences inside dollar-quoted arguments (`\x` in `$'\x48'`)
+* `function-definition` - function name(s) when defining a function (`f` and `g` in `f g () { : }`)
```

Comment:
Test cases (written before I read the diff):

`function { pwd }` — anonymous function

`function pwd` — defines a function named `pwd`, _not_ runs that command

`function >/dev/null { pwd }` — anonymous function

`function f >/dev/null { pwd }` — named function with sticky redirection

`function >/dev/null f { pwd }` — syntax error (!)

The above also support an optional `()` after the function names and before the brace.

`() {}` — anonymous function

`() pwd` — anonymous function

`() () () pwd` — anonymous function for calling tor

The above also come in named variants.

`function () () pwd` — two anonymous functions

`function f () () pwd` — named function whose body is an anonymous function

`function () f () pwd` — anonymous function that defines a function named `f`

`function () () f pwd` — two anonymous functions, `f` is in command position (unknown-token)

Then there's the MULTI_FUNC_DEF variants of all these.

Then there's

`unsetopt multifuncdef; function f g h { pwd }` — creates a function named `f`.  Probably a zsh bug, actually, but shouldn't we highlight `g` and `h` as errors nevertheless?  (I saw the test for the non-`function`-syntax variant of this.)


Question: In `() pwd`, should the `()` be highlighted as a command position?  What about `function { pwd }`?

Reply:
Another case (github didn't let me edit comment after I pressed the green button):

```zsh
% function f
pwd
% f
```

Reply:
How about highlighting it the same way as the newlines in 

```zsh
for 1 in foo bar baz
do
  echo $1
done
```

?